src/main/java/com/diffblue/deeptestutils/Reflector.java

```diff
@@ -96,12 +88,11 @@ public static void setField(final Object obj, final String fieldName,
    * @exception IllegalAccessException if an error occurs
    */
   public static <T> void setField(final Class<T> c, final Object o,
-                                   final String fieldName, final Object newVal)
-      throws NoSuchFieldException, IllegalArgumentException,
-             IllegalAccessException {
+                                  final String fieldName, final Object newVal) {
 
     if (c == null) {
-      throw new NoSuchFieldException();
+      final NoSuchFieldException e = new NoSuchFieldException();
+      throw new DeeptestUtilsRuntimeException(e.getMessage(), e.getCause());
```

Comment:
Maybe add a more useful message like "Class of the field to be set cannot be null." I think also you can just put the `new NoSuchFieldException()` directly into the cause and get rid of the first line.

Reply:
Done.